tests/conftest.py

```diff
@@ -366,3 +1223,681 @@ def pytket_nested_conditionals_circuit(
+            name="then0",
+            succs=["continue0"],
+            preds=["entry"],
+            composition=["then0", "then1", "then2"],
+            visited=False,
+        ),
+        "continue0": Block(
+            name="continue0",
+            succs=[],
+            preds=["then0", "else0"],
+            composition=["continue0"],
+            visited=False,
+        ),
+    }
+    return cfg
+
+
+@fixture
+def collapsed_jump_right_cfg() -> dict:
+    cfg = {
+        "entry": Block(
+            name="entry",
+            succs=["then0", "else0"],
+            preds=[],
+            composition=["entry"],
+            visited=False,
+        ),
+        "then0": Block(
+            name="then0",
+            succs=["continue0"],
+            preds=["entry"],
+            composition=["then0"],
+            visited=False,
+        ),
+        "else0": Block(
+            name="else0",
+            succs=["continue0"],
+            preds=["entry"],
+            composition=["else0", "then1", "then2"],
+            visited=False,
+        ),
+        "continue0": Block(
+            name="continue0",
+            succs=[],
+            preds=["else0", "then0"],
+            composition=["continue0"],
+            visited=False,
+        ),
+    }
+    return cfg
+
+
+@fixture
+def collapsed_complex_chain() -> dict:
+    cfg = {
+        "entry": Block(
+            name="entry",
+            succs=["else0", "then0"],
+            preds=[],
+            composition=["entry"],
+            visited=False,
+        ),
+        "else0": Block(
+            name="else0",
+            succs=["continue0"],
+            preds=["entry"],
+            composition=["else0"],
+            visited=False,
+        ),
+        "then0": Block(
+            name="then0",
+            succs=["continue0"],
+            preds=["entry"],
+            composition=["then0", "then1", "then2"],
+            visited=False,
+        ),
+        "continue0": Block(
+            name="continue0",
+            succs=["leftbr1", "rightbr1"],
+            preds=["then0", "else0"],
+            composition=["continue0", "continue1", "continue2"],
+            visited=False,
+        ),
+        "leftbr1": Block(
+            name="leftbr1",
+            succs=["exit1"],
+            preds=["continue0"],
+            composition=["leftbr1"],
+            visited=False,
+        ),
+        "rightbr1": Block(
+            name="rightbr1",
+            succs=["exit1"],
+            preds=["continue0"],
+            composition=["rightbr1", "rightbr2"],
+            visited=False,
+        ),
+        "exit1": Block(
+            name="exit1",
+            succs=[],
+            preds=["rightbr1", "leftbr1"],
+            composition=["exit1", "exit2"],
+            visited=False,
+        ),
+    }
+    return cfg
+
+
+@fixture
+def collapsed_nested_chains() -> dict:
+    cfg = {
+        "entry": Block(
+            name="entry",
+            succs=["else0", "then0"],
```

Comment:
What are the conditions for the following nodes? Or is that not relevant in this data structure?